kaldb/src/main/java/com/slack/kaldb/chunk/ReadOnlyChunkImpl.java

```diff
@@ -99,22 +99,21 @@ public ReadOnlyChunkImpl(
         Executors.newSingleThreadExecutor(
             new ThreadFactoryBuilder().setNameFormat("readonly-chunk-%d").build());
     this.searchContext = searchContext;
-    String slotName = String.format("%s-%s", searchContext.hostname, slotId);
+    this.slotName = String.format("%s-%s", searchContext.hostname, slotId);
 
+    this.cacheSlotMetadataStore = cacheSlotMetadataStore;
     this.replicaMetadataStore = replicaMetadataStore;
     this.snapshotMetadataStore = snapshotMetadataStore;
     this.searchMetadataStore = searchMetadataStore;
 
-    // todo - remove the unnecessary additional directory
-    String cacheSlotPath = String.format("%s/%s", CACHE_SLOT_STORE_ZK_PATH, slotName);
-    cacheSlotMetadataStore =
-        new CacheSlotMetadataStore(metadataStoreService.getMetadataStore(), cacheSlotPath, true);
-    cacheSlotMetadataStore.addListener(cacheNodeListener());
-
     CacheSlotMetadata cacheSlotMetadata =
         new CacheSlotMetadata(
-            METADATA_SLOT_NAME, Metadata.CacheSlotState.FREE, "", Instant.now().toEpochMilli());
-    cacheSlotMetadataStore.create(cacheSlotMetadata);
+            slotName, Metadata.CacheSlotState.FREE, "", Instant.now().toEpochMilli());
+    cacheSlotMetadataStore.createSync(cacheSlotMetadata);
+
+    CacheSlotMetadataStore cacheSlotNodeMetadataStore =
```

Comment:
I think updating the classdoc to explain the difference between `cacheSlotNodeMetadataStore` and `cacheSlotMetadataStore` would make it easier to read the code at a later time.

Nit: The names are too close. Consider using a different name for `cacheSlotNodeMetadataStore` like `slotListener`.

Reply:
Classdoc added for each constructor explaining the structure and intended use case for each, and renamed the `cacheSlotNodeMetadataStore` variable to better reflect it's for use as a listener only.